documentation/release_notes.md

```diff
@@ -0,0 +1,28 @@
+<!--Title:Release Notes-->

```

Comment:
@mysticmind I was also thinking whether we should do just Migration Notes or provide full release notes. I personally believe that having descriptive release notes in single place that people can look on might be useful. I think that we can do that for each release even if it doesn’t contain breaking changes.

I’m also fine with keeping it as simple migration guide for the cases that it’s needed. It’s just decision about the approach that we’d like to go.

Regarding other comments - I’m totally fine, I’ll update the changes :)

Thanks for feedback!

Reply:
For the current thing at hand, we can just do migration notes. 

For new releases, we can plan to provide full release notes using the GitHub Release notes feature. I would suggest to have a Release Notes page in documentation with links pointing to the GitHub Release Notes.

Reply:
@mysticmind that’s also good option. 
@jeremydmiller @jokokko - your thought?

Reply:
Also if we are planning to capture breaking changes, new functionality etc which we want to highlight to users, it has to be in appropriate sections within the existing documentation pages.

Reply:
Fully agreed. This should be only summary and reference to proper documentation pages. We should fall into a trap taking shortcut with putting information here instead of doing proper documentation with samples etc. :)

Reply:
I really like how Vue.js guys maintain GitHub release notes i.e. https://github.com/vuejs/vue/releases

Reply:
@mysticmind Thanks for sample! Yes I’ve release notes looks nice and tidy. I’ll also have a look on other open source project. Then we could discuss with separate issue exact format that we’d like to have.

Reply:
@oskardudycz My wish is that tags are used with each release and in a consistent manner (down to naming convention). Then it's easy even locally to do something like git changelog (from git extras) with start & final tags. Regarding summaries of changes, esp. breaking, I'd expect to find them under GH release page.

Reply:
@mysticmind @jokokko I suggest to move detailed discussion about release notes and changelog to separate github issue (I'll create one). For now let's decide if we'd like to keep the release notes/migration guide as was placed in this PR. I wouldn't like to block the whole feature because of the discussion about it as @amferguson is waiting for a fix.

If we can't conclude if it's fine or not, then I think that the most reasonable would be to move documentation changes out of this PR and then having final decision create specific PR.

Reply:
@oskardudycz Agree that we can discuss around the release notes in a separate issue. Let us go ahead and merge the PR with the following changes:
1. call the documentation file as "migration_guide.md"
2. change the content to remove any references to "release notes" as a text.

Reply:
Fine for me

Reply:
@mysticmind I applied your comments besides Readme. I left it for the final decision about the Releases management.